src/main/resources/react/src/pages/UserInfo/tab_pages/my_profile/myProfile.js

```diff
@@ -1,176 +1,184 @@
 	putUserInfoAsync
 } from '../../../../redux/profile/profileAction';
 
+import { getTranslatedText } from '../../../../components/local/localisation';
+
+import './myProfile.scss';
+
 
 const MyProfile = () => {
 	const dispatch = useDispatch();
-	const { profile } = useSelector( state => state.profileMe );
-
-	const addInputMe = useCallback( ( e ) => {
-		dispatch( addMeInputValue( { [e.target.name]: e.target.value } ) );
+	const { lang } = useSelector( state => state.auth );
+	const {
+		firstName,
+		lastName,
+		phones,
+		mockPhones,
+		children,
+		mockChildren,
+		prohibitionToSendAboutMe,
+		prohibitionToSendChildren,
+		receivedChildrenFromBack
+	} = useSelector( state => state.profileMe );
+
+	const phonesArray = !phones.length ? mockPhones : phones;
+	const childrenArray = !children.length ? mockChildren : children;
+
+
+	useEffect( () => {
+		dispatch( fetchUserInfoAsync() );
 	}, [ dispatch ] );
 
-	const addInputCity = ( e ) => {
-		console.log( { [e.target.name]: e.target.value } );
-	};
+	useEffect( () => {
+		dispatch( changePermissionProfile() );
+	}, [ dispatch, firstName, lastName, phones ] );
+
+	useEffect( () => {
+		dispatch( changePermissionChildren() );
+	}, [ dispatch, children ] );
 
-	const addInputPhone = useCallback( ( e, id ) => {
-		dispatch( addPhoneInputValue( { [e.target.name]: e.target.value }, id ) );
+	const changeInput = useCallback( ( event ) => {
+		dispatch( changeInputValue( [ event.target.name, event.target.value ] ) );
 	}, [ dispatch ] );
 
-	const phoneAdd = useCallback(()=> {
-		dispatch(addPhone())
-	},[dispatch])
+	const changeInputPhone = useCallback( ( event, id ) => {
+		dispatch( changePhoneInputValue( [ event.target.name, event.target.value ], id ) );
+	}, [ dispatch ] );
 
-	const phoneRemove = useCallback(()=> {
-		dispatch(deletePhone())
-	},[dispatch])
+	const phoneAdd = useCallback( () => {
+		dispatch( addPhone() );
+	}, [ dispatch ] );
 
-	const showPhone = useCallback( ( value, id ) => {
-		dispatch( addPhoneInputValue( { show: !value }, id ) );
+	const phoneRemove = useCallback( () => {
+		dispatch( deletePhone() );
 	}, [ dispatch ] );
 
-	const addInputChildren = useCallback( ( e, id ) => {
-		dispatch( addChildrenInputValue( { [e.target.name]: e.target.value }, id ) );
+	const changeInputChildren = useCallback( ( event, id ) => {
+		dispatch( changeChildrenInputValue( [ event.target.name, event.target.value ], id ) );
 	}, [ dispatch ] );
 
 	const pickBoyOrGirl = useCallback( ( id, gender ) => {
-		dispatch( addChildrenInputValue( { sex: gender }, id ) );
+		dispatch( chooseGender( gender, id ) );
 	}, [ dispatch ] );
 
-	const childDelete = useCallback( ( childrenId ) => {
-		dispatch( deleteChild( childrenId ) );
-	}, [ dispatch ] );
 
-	const childAdd = useCallback( id => {
+	const childAdd = useCallback( () => {
 		dispatch( addChild() );
 	}, [ dispatch ] );
 
+
+	const childDelete = useCallback( ( idx, childrenId, localDelete ) => {
+		if (localDelete) {
```

Comment:
You could make it prettier with ternare operator. Kinda:
```
localeDelete
    ? dispatch( deleteChild( [ idx, childrenId, localDelete ] ) )
    :  dispatch( deleteChildrenAsync( [ idx, childrenId ] ) );
```
But it's just a recomendation.

Reply:
I fixed it with my hands. I don't now how to do it  automatically
I mean. I found it in the Webstorm, but it doesn't apply

Reply:
I did it manually. It's not so big deal. 😃